contracts/contracts/FillManager.sol

```diff
@@ -142,6 +142,11 @@ contract FillManager is Ownable, LpWhitelist {
     ) external {
         require(l1Resolver != address(0), "Resolver address not set");
         require(fills[requestId] != fillId, "Fill valid");
+        require(
+            fillId != generateFillId(),
+            "Cannot invalidate block hash of previous block"
```

Comment:
Nitpick: since we talk about invalidating fills, not block hashes, in order to be more consistent I'd reword the message to be more like
```
         "Cannot invalidate fill within same block"
```